doc/user_guide/user_guide.md

```diff
@@ -34,120 +29,14 @@ Examples of valide error codes:
-about it.
-   
-#### parameter
-This attribute takes the placeholder name and the parameter value 
-that will replace this placeholder as argument. It can be used for both message 
-and mitigations. For example usages, please see the 
-[Parameters](#Parameters) section.
-
-## Usage 
-
-### Simple Messages
-```python
-ExaError.message_builder("E-TEST-1").message("Something went wrong.")
-```
-The result of the error message:
-```
-E-TEST-1: Something went wrong.
-```
-
-
-### Parameters
-You can specify placeholders in the message and replace them with 
-parameters values.
-
-```python
-ExaError.message_builder("E-TEST-2")
-    .message("Unknown input: {{input}}.")
-    .parameter("input", "unknown", "The illegal user input.")
-```
-or inline: 
-```python
-ExaError.message_builder("E-TEST-2")
-    .message("Unknown input: {{input}}.",  "unknown")
-```
-The result of both error messages is same:
-```
-E-TEST-2: Unknown input: 'unknown'.
-```
-
-The optional third argument for `parameter(name, value, description)` 
-method is used to generate a parameter description for the error-catalog.
-
-The builder automatically quotes parameters (depending on the type of the 
-parameter). If you don't want that, use the `|uq` suffix in the corresponding 
-placeholder, as follows:
-
-```python
-ExaError.message_builder("E-TEST-2")
-    .message("Unknown input: {{input|uq}}.")
-    .parameter("input", "unknown", "The illegal user input.")
-```
-The result of the error message:
-```
-E-TEST-2: Unknown input: unknown.
-```
-
-### Mitigations
-The mitigations describe actions the user can take to resolve the error. 
-Here is an example of a mitigation definition:
-
-```python
-ExaError.message_builder("E-TEST-2")
-    .message("Not enough space on device.")
-    .mitigation("Delete something.")
-```
-The result of the error message:
-```
-E-TEST-2: Not enough space on device. Delete something.
-```
-
-You can use parameters in mitigations too.
-```python
-ExaError.message_builder("E-TEST-2")
-    .message("Not enough space on device {{device}}.")
-    .mitigation("Delete something from {{device}}.")
-    .parameter("device", "/dev/sda1", "name of the device")
-```
-The result of the error message:
-```
-E-TEST-2: Not enough space on device '/dev/sda1'. Delete something from '/dev/sda1'.
-```
-
-You can chain `mitigation` definitions if you want to tell the users that there 
-is more than one solution, as follows:
-```python
-ExaError.message_builder("E-TEST-2")
-    .message("Not enough space on device.")
-    .mitigation("Delete something.")
-    .mitigation("Create larger partition.")
-```
-The result of the error message:
-```
-E-TEST-2: Not enough space on device. Known mitigations:
-* Delete something.
-* Create larger partition.
-```
\ No newline at end of file
+#### parameters
+This parameter takes a dictionary of placeholder names and the respective parameter values.
+They will be used to replace the placeholders in the mitigations and messages.
```

Comment:
```suggestion
#### parameters
This argument takes a dictionary of placeholder names and the respective parameter values.
They will be used to replace the placeholders in the mitigations and messages.
```
Use a different word to avoid confusion. Also an example wouldn't hurt, even if the readme already contains examples.